### ISSUE TITLE
Is/should it be possible with a future `requestedDateTime` when posting an order?

### ISSUE THREAD

**Issue by joakimbeng** [on-api/on-api#93]
The documentation doesn't explicitly say that [`requestedDateTime`](https://github.com/on-api/on-api/blob/master/2.5.0/spec/orders.md#requesteddatetime) can be a date in the future, but I guess it should be possible with that. What else would it be for?

Why I'm asking is because the [`operationalState`](https://github.com/on-api/on-api/blob/master/2.5.0/common/dataformats.md#operationalstate) field on a subscription can only have two values (`ACTIVATED` or `SUSPENDED`) according to the spec. So if a SP orders a service with a future `requestedDateTime` then neither of the two possible operationalState values correctly represents the state of the service as it's not really activated yet, only "scheduled" for activation.

Or is this by design and intention? I.e. should the operationalState also be `ACTIVATED` in the case of a future requestedDateTime? If so I think the documentation should be updated with a note on this, and if not we should add an additional possible operationalState like `SCHEDULED` or similar. What do you think?

**Comment by joakimbeng** [comment]
Our (iTUX) current implementation decision is that we only create subscriptions when a service is actually activated, so for orders with a future `requestedDateTime` we don't even respond with a `subscriptionId`.

This means that in all of the [orders examples](https://github.com/on-api/on-api/blob/master/2.4.0/spec/orders.md) where `state=RECEIVED` and `operation=ACTIVATE` we will not return a `subscriptionId`.

I suggest that the documentation is updated to reflect this as it's the most reasonable thing to do, unless we introduce another `operationalState` like `PENDING` or similar.

**Comment by teemufrosen** [comment]
I agree with introducing a new field for subscriptions which are not yet activated (PENDING/SCHEDULED). Regarding the creation of a subscription before its activation date, that's a bit trickier and needs further discussion. Will bring this up on next meeting.

**Comment by Xake** [comment]
As we see it we do not think there needs to be a new field.
As we see it, orderId is for tracking (and canceling) orders, and is enough for tracking the order until it results in a subscription.
 
What we feel is problematic is  that according to the current documentation, you should always respond with a subscriptionId even before we have reached a state when there is a need for a subscription.
 
Say for example you send an order. We respond with RECEIVED.
Before we even have start to handle the order in our system, you send a DELETE on the orderId to cancel the order.
Or say that our business logic results in us having to mark the order as DONE_FAILED due to us being unable to create a service from the order.
 
According to the documentation we should already respond with a subscriptionId when we respond with RECEIVED, even when it is something that does not actually results in a subscription that you can or need to interact with.
 
As such, we want to delay the creation and thus the need to repond with a subscrioptionId until there is an actual need for a subscription that can be interacted with.
